.github/workflows/tests.yaml

```diff
@@ -0,0 +1,435 @@
+name: E2E Tests for Astro Deploy Action
+
+on:
+  push:
+    branches:
+      - main
+      - e2e-tests
+  workflow_dispatch:
```

Comment:
Can we allow manual runs to override the secrets? We might have a dev deployment we want to run the tests against as a once off, or we might want to run these tests per release on both hosted and hybrid organizations, etc.

Reply:
let me see if I could add that cleanly, without exposing existing secrets

Reply:
Added support to specify custom inputs when triggered manually